packages/button/src/Button/components/GenericButton.spec.tsx

```diff
@@ -86,3 +86,48 @@ describe("<GenericButton />", () => {
     expect(screen.getByTestId("custom-prop")).toBeInTheDocument()
   })
 })
+
+describe("<GenericButton /> with native HTML `form` attributes", () => {
+  const buttonFormAttributes = {
+    form: "linked-form-id",
+    formAction: "/",
+    formMethod: "post",
+    formEncType: "text/plain",
+    formTarget: "_self",
+    formNoValidate: false,
+  } satisfies ButtonFormAttributes
+
+  it("will pass form props into the button", () => {
+    render(
+      <GenericButton
+        label="submit button"
+        data-testid="custom-prop"
+        {...buttonFormAttributes}
+      />
+    )
+
+    expect(screen.getByRole("button", { name: "submit button" })).toBeTruthy()
+    expect(
+      screen.getByRole("button", { name: "submit button" })
+    ).toHaveAttribute("form", buttonFormAttributes.form)
+  })
+
+  it("will sumbit a form linked by the id", () => {
+    const onSubmit = jest.fn(e => e.preventDefault())
+    render(
+      <div>
+        <form id="linked-form-id" onSubmit={onSubmit}>
+          <input type="text" defaultValue="test" />
+        </form>
+        <GenericButton
+          label="submit button"
+          data-testid="custom-prop"
+          type="submit"
+          {...buttonFormAttributes}
+        />
+      </div>
+    )
+    fireEvent.click(screen.getByRole("button", { name: "submit button" }))
+    expect(onSubmit).toHaveBeenCalledTimes(1)
```

Comment:
I like the test and would definitely leave it, but you are essentially testing native `button` functionality (ie. it submitting the form in the `form` attribute), which some people would say is superfluous.

Reply:
Indeed, that person is me 😄

Let's not make it a habit to add superfluous tests. Lest someone returns later and is confused as to why this is tested thinking it's for something important

Reply:
Honestly, I was on the fence about doing this, I wanted to validate that the prop would actually work if it was passed into the component, but I am testing essentially to test that HTML should work as HTML 🤔 So maybe just its existence is enough?